registry/auth.go

```diff
@@ -147,18 +152,44 @@ func decodeAuth(authStr string) (string, string, error) {
 
 // load up the auth config information and return values
 // FIXME: use the internal golang config parser
-func LoadConfig(rootPath string) (*ConfigFile, error) {
-	configFile := ConfigFile{Configs: make(map[string]AuthConfig), rootPath: rootPath}
-	confFile := path.Join(rootPath, CONFIGFILE)
+func LoadConfig(configDir string) (*ConfigFile, error) {
+	if configDir == "" {
+		configDir = filepath.Join(homedir.Get(), ".docker")
+	}
+
+	configFile := ConfigFile{
+		AuthConfigs: make(map[string]AuthConfig),
+		filename:    filepath.Join(configDir, CONFIGFILE),
+	}
+
+	// Try happy path first - latest config file
+	if _, err := os.Stat(configFile.filename); err == nil {
+		data, err := ioutil.ReadFile(configFile.filename)
+		if err != nil {
+			return &configFile, err
+		}
+		err = json.Unmarshal(data, &configFile)
```

Comment:
The usernames/passwords are getting cleared out later because the config value is simply returned here without breaking apart the encoded values back into username and password. Later when the config is saved again, the Auth value is set, but Auth gets overridden using `encodeAuth(&authCopy)`, which encodes using the now empty username and password.

This probably needs to be done here for each value
```
authConfig.Username, authConfig.Password, err = decodeAuth(authConfig.Auth)
if err != nil {
	return &configFile, err
}
authConfig.Auth = ""
```